src/baseline.rs

```diff
@@ -340,3 +363,105 @@ fn need_retry(res: Result<(), crates_index::Error>) -> anyhow::Result<bool> {
         Err(err) => Err(err.into()),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crates_index::{Crate, Version};
+
+    use super::choose_baseline_version;
+
+    fn new_mock_version(version_name: &str, yanked: bool) -> Version {
+        // `crates_index::Version` cannot be created explicitly, as all its fields
+        // are private, so we use the fact that it can be deserialized.
+        serde_json::from_value(serde_json::json!({
+            "name": "test-crate",
+            "vers": version_name,
+            "deps": [],
+            "features": {},
+            "yanked": yanked,
+            "cksum": "00".repeat(32),
+        }))
+        .expect("Failed to create crates_index::Version from a hand-written JSON.")
```

Comment:
@obi1kenobi can you describe what exactly should be the format of `expect` messages? I remember that you said that they should start with a small letter and not end with a comma. But also, here it's explained that the string should describe an invariant, not what went wrong: https://stackoverflow.com/questions/66362625/why-is-rusts-expect-called-expect

Reply:
This is worth a read: https://doc.rust-lang.org/std/error/index.html#common-message-styles

Definitely lowercase and not ending with punctuation. For the rest, read the doc at the link above and then use your best judgment :)

Reply:
Changed to lowercase without ending punctuation. I used the “expect as precondition” style everywhere, hope it's finally good enough ;)